test/functional/page_objects/common_page.ts

```diff
@@ -411,9 +411,18 @@ export class CommonPageObject extends FtrService {
    * Clicks cancel button on modal
    * @param overlayWillStay pass in true if your test will show multiple modals in succession
    */
-  async clickCancelOnModal(overlayWillStay = true) {
+  async clickCancelOnModal(overlayWillStay = true, ignorePageLeaveWarning = false) {
     this.log.debug('Clicking modal cancel');
-    await this.testSubjects.click('confirmModalCancelButton');
+    await this.testSubjects.exists('confirmModalTitleText');
+
+    await this.retry.try(async () => {
```

Comment:
we have to wait for the modal to fade in, which takes time, likely even longer in CI